build/ci/community/wait_demo_pr_checks.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -xe
+set +o pipefail
+
+wait_for_checks_to_complete(){
```

Comment:
Let's split this function to 3 smaller functions:

1. wait_for_tests_to_finish
2. collect_tests_results
3. wait_for_summary_to_finish